Makefile

```diff
@@ -13,3 +13,12 @@ lint: ## Run ruff linting
 .PHONY: format
 format: ## Run ruff formatting
 	@ruff format refl1d/ tests/ setup.py
+
+.PHONY: clean
+clean: ## Delete some cruft from builds/testing/etc.
+	rm -f `find . -type f -name '*.py[co]'`
+	rm -rf `find . -name __pycache__ -o -name "*.egg-info"` \
+		`find . -name 'output-*'` \
+		.coverage build dist doc/_build \
+		.pytest_cache \
+		.ruff_cache 
```

Comment:
Will this work on windows? [Does it matter?]

Are we sure we cannot have spaces in any of the found names?

Reply:
referencing line 21 specifically? 
this is just to clean up the files written out by running `pytest` locally:
```
output-expt.json
output-profile.dat
output-refl.dat
output-slabs.dat
output-steps.dat
```

I'm not sure that this would work on windows, which does matter for developers on windows. i'm not familiar with cmd / powershell or how exactly to do os checks in a Makefile, but if you have ideas we can expand on this.

Reply:
Why is there a Makefile in the root folder of the project now?  It does not build the project - can it live somewhere else?  Having documentation (this file exists to do <...things>) in a comment at the top of the Makefile would be helpful, too.

Reply:
root is a pretty common home for Makefiles, especially ones that contain convenience functions, in this case cleaning up files (many of which are in or relative to the root directory), as well as formatting and linting. 

having it here also makes it extensible if we do find we want to add convenience targets for building, installing, running in dev mode, compiling the web client, etc etc. 

we could add documentation at the top, but typically i just define a `help` target which prints out the targets contained and what they do, [for example](https://github.com/monarch-initiative/monarch-app/blob/main/Makefile#L8) (although I've recently grown fond of the much slimmer one-liner which grabs comments with double-# and uses those)

Reply:
For the actions that require significant path manipulations, we could make a separate `bash` script and call that from the Makefile.  

`bash` is available on Windows with `conda install -c conda-forge git-bash` or by installing git system-wide, and importantly `bash` is available on Windows Github action runners by default.  We could make it a requirement for a dev environment on Windows?

Reply:
We could yeah. I do tend to prefer root level, as its pretty natural, common, and transparent/easy to find, read, extend, etc. But we totally can if you think it makes sense for this project.

Reply:
what's the verdict on this btw - ok to leave it top level, or prefer to move into a subdir?